src/arrayMethodJoin.js

```diff
@@ -4,8 +4,25 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+    let element;
+
+    for (let i = 0; i < this.length; i++) {
+      if (i > 0) {
+        result += separator;
+      }
+
+      element = this[i];
+
+      if ((element === undefined) || (element === null)) {
+        result += '';
+      } else {
+        result += element;
+      }
```

Comment:
in order to avoid adding `else`, you could place `element` declaration inside the loop